sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -427,6 +427,13 @@ private[hive] case class MetastoreRelation
           .getOrElse(sqlContext.defaultSizeInBytes))
     }
   )
+  override def sameResult(plan: LogicalPlan): Boolean = {
+      plan match {
+           case mr: MetastoreRelation =>
```

Comment:
When you open the PR make sure the base branch says `master` not `branch-1.X`.  https://help.github.com/articles/comparing-commits-across-time/#comparing-branches